x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/signals/open_close_signals.test.ts

```diff
@@ -4,134 +4,134 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ServerInjectOptions } from 'hapi';
 import { DETECTION_ENGINE_SIGNALS_STATUS_URL } from '../../../../../common/constants';
-import { setSignalsStatusRoute } from './open_close_signals_route';
-import * as myUtils from '../utils';
-
 import {
   getSetSignalStatusByIdsRequest,
   getSetSignalStatusByQueryRequest,
   typicalSetStatusSignalByIdsPayload,
   typicalSetStatusSignalByQueryPayload,
   setStatusSignalMissingIdsAndQueryPayload,
 } from '../__mocks__/request_responses';
-import { createMockServer, createMockConfig, clientsServiceMock } from '../__mocks__';
+import { requestContextMock, serverMock, requestMock } from '../__mocks__';
+import { setSignalsStatusRoute } from './open_close_signals_route';
 
 describe('set signal status', () => {
-  let server = createMockServer();
-  let config = createMockConfig();
-  let getClients = clientsServiceMock.createGetScoped();
-  let clients = clientsServiceMock.createClients();
+  let server: ReturnType<typeof serverMock.create>;
+  let { clients, context } = requestContextMock.createTools();
 
   beforeEach(() => {
-    // jest carries state between mocked implementations when using
-    // spyOn. So now we're doing all three of these.
-    // https://github.com/facebook/jest/issues/7136#issuecomment-565976599
-    jest.resetAllMocks();
-    jest.restoreAllMocks();
-    jest.clearAllMocks();
-    jest.spyOn(myUtils, 'getIndex').mockReturnValue('fakeindex');
-
-    server = createMockServer();
-    config = createMockConfig();
-    getClients = clientsServiceMock.createGetScoped();
-    clients = clientsServiceMock.createClients();
-
-    getClients.mockResolvedValue(clients);
-    clients.clusterClient.callAsCurrentUser.mockResolvedValue(true);
-
-    setSignalsStatusRoute(server.route, config, getClients);
+    server = serverMock.create();
+    ({ clients, context } = requestContextMock.createTools());
+
+    clients.clusterClient.callAsCurrentUser.mockResolvedValue(true); // TODO: what is the return value of a successful update?
```

Comment:
Is this attached commented TODO still valid?

Reply:
Same as above: since we just pass this result directly into the response we're exercising the happy path here. But these TODOs were intentional and still valid, and I'll figure out how to mock this call appropriately.

Reply:
7087da2905c95edd80aa73c4f1a90005b7b8fe83